### PR TITLE
Add step system

### LINKED ISSUE #18: Rename log/logger/logging to console/output
We're not doing logging but console outputs for command line interfaces, so the current terminology is misleading.

### LINKED ISSUE #19: Add step management
Add a way for the console to print nice outputs like such for step-by-step processes:

```go
Accessing database...ok
Checking database integrity...ok
Synchronizing local store...ok
  > Local store up to date with remote database
Installation in progress...ok
  > Installing with current database
  > Connection to 172.187.10.23 secure
  > Found 36 dependency requirements
  > Dependencies resolved
```

With a user-friendly interface.

### LINKED ISSUE #20: Add global prompt
Similar to #17 but for the prompter.

### PR DESCRIPTION
**Pull request by Ullaakut**
## Goal of this PR

This PR does the following:

- Documentation improvements
- Renaming of every occurrence of `log`, `logger` and `logging` to `console` and `output` depending on the context
- Renaming of the `disgo/logger` package to `disgo/console`
- Adds methods to the disgo console
	- StartStep
	- StartStepf
	- EndStep
	- FailStep
	- FailStepf
- Makes the disgo prompter usable as a global prompter
- Removes old examples for easier maintenance
	- We can add more examples again once the library becomes stable and is ready for a 1.0 release
- ~In the README, shows an example of code next to the three following screenshots. Might not be the best solution but I'm not sure how to show examples of what our library does in a concise but informative manner 🤔Ideas @jlevesy @skwair~ ?
	- In the end, I rewrote the documentation. [WDYT about it](https://github.com/Ullaakut/disgo/blob/add-step-system/README.md)?

Fixes #18
Fixes #19
Fixes #20

## Screenshots

<p align="center">
example_step_by_step

</p>

## TODO

- [x] Add unit tests (might be done later due to time constraints)
	- [x] In the end, had time write tests to cover 100% of the `console` and `prompter` package

## How to test it

- [ ] `go test ./...`
- [ ] Play around with the example
	- [ ] Try queueing debug logs in a task while debug is disabled
	- [ ] Try queuing error logs
	- [ ] Try using different methods for queuing logs
	- [ ] Try exiting a program without calling `EndStep` or `FailStep`
- [ ] Read the code
- [ ] [Take a look at the README in GitHub (Not in Markdown)](https://github.com/Ullaakut/disgo/blob/add-step-system/README.md)

## Additional notes

<p align="center">
	
</p>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,167 +18,260 @@
...
+    // Debug methods are similar to info, except that they are not printed
+    // if debug outputs are not enabled on the console.
+    myConsole.Debugln("Number of days in a year:", 365)
+    myConsole.Debugf("Number of days in a year: %d\n", 365)
+    myConsole.Debug("Number of days in a year: 365\n")
+
+
+    // Error methods are similar to info, except that they are written on
+    // the error writer (os.Stderr by default).
+    myConsole.Debugln("Number of days in a year:", 365)
```
Comment (dtomcej) [inline README.md]: Error methods are referenced, but Debug methods are shown.

File: README.md [hunk 2]
```diff
@@ -18,167 +18,260 @@
...
+    // Debug methods are similar to info, except that they are not printed
+    // if debug outputs are not enabled on the console.
+    console.Debugln("Number of days in a year:", 365)
+    console.Debugf("Number of days in a year: %d\n", 365)
+    console.Debug("Number of days in a year: 365\n")
+
+
+    // Error methods are similar to info, except that they are written on
+    // the error writer (os.Stderr by default).
+    console.Debugln("Number of days in a year:", 365)
```
Comment (dtomcej) [inline README.md]: Same as above.

**Review by dtomcej — CHANGES_REQUESTED** [review]
Awesome job! 👏

**Review by jlevesy — APPROVED** [review]
LGTM :+1:

**Pull request merged**